### ISSUE TITLE
Hyphenated environment variables

### ISSUE THREAD

**Issue by samuele-artuso** [Financial-Times/n-heroku-tools#489]
**TL;DR**
Bug: `next-router` won't run some services locally because it's using non-standard environment variable naming.

As far as I know the bug seems to affect only local Linux environments.

I'm proposing a fix for which I issued 2 PRs (https://github.com/Financial-Times/n-heroku-tools/pull/490 and https://github.com/Financial-Times/next-router/pull/128).

**The long story**
`n-heroku-tools` uses environment variables to tell `next-router` which services need to be routed locally. Such environment variables are named after services and hold the number of the local port on which they need to be rerouted ([code](https://github.com/Financial-Times/n-heroku-tools/blob/master/tasks/run.js#L131)).

The name of the service is taken _almost_ as is from its own `package.json` (the bits `ft-` and `next-` at the beginning of the name will be stripped, if present) ([code](https://github.com/Financial-Times/n-heroku-tools/blob/master/lib/normalize-name.js#L17)).

Given our naming conventions, these environment variables will almost always be lowercase strings, that may contain one or more hyphens.

E.g.

Service name   | Resulting environment variable name
-------------- | -----------------------------------
`next-article` | `article`
`next-session` | `session`
`next-front-page` | `front-page`
`next-lure-api` | `lure-api`

As can be seen in the last 2 examples, this simple algorithm will lead `nht run` to invoke the router using **hyphenated environment variable names**.

Albeit they may work in some circumstances, they shouldn't generally be relied upon because they are not treated consistently across the ecosystem of Linux/Unix tools ([see why](https://www.google.co.uk/search?q=hyphenated+environment+variable+names&oq=hyphenated+environment+variable+names&aqs=chrome..69i57.4759j0j7&sourceid=chrome&ie=UTF-8)).

I experienced an issue on my Linux laptop (running Ubuntu 17.10 Artful Aardvark) caused by this. I was running the `lure-api` locally, but all requests were still being routed over to Heroku. The reason is that my router can't seem to see those hyphenated environment variables.

I managed to get hold of another developer in the company using Ubuntu (@ulpian), and he confirmed this bug.

I'm proposing a fix for which I'm issueing 2 pull requests (https://github.com/Financial-Times/n-heroku-tools/pull/490 and https://github.com/Financial-Times/next-router/pull/128).

Apart from sorting out developers running Linux, the business value of this fix is that it implements a standard approach that should future-proof `next-router`.

I'm being cautious so I kept backward compatibility—that is, the old behaviour is still there and it takes precendence over my fix.

**Comment by simonplend** [comment]
Fixed in #490
